src/Domains/Subscription/Prices/Repositories/PriceRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Subscription\Prices\Repositories;
+
+use Baka\Contracts\AppInterface;
+use Baka\Traits\SearchableTrait;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Kanvas\Exceptions\ModelNotFoundException as ExceptionsModelNotFoundException;
+use Kanvas\Subscription\Prices\Models\Price;
+
+class PriceRepository
+{
+    use SearchableTrait;
+
+    public static function getModel(): Model
+    {
+        return new Price();
+    }
+
+    public static function getByIdWithApp(int $id, AppInterface $app): Price
+    {
+        try {
+            return self::getModel()::notDeleted()
```

Comment:
You can do it different 
$price = Kanvas\Subscription\Prices\Models\Price::getById($id, $app);
$plan = $price->plan;

In Laravel , when you use eloquent queries, all relationships are retrieved in the result, in this case plan, and you can access to it with maggic attribute plan
 
 Also our models has default relationships

-  user
- company
- app

Reply:
When using companies subscriptions to apps, you can have prices/plans without a relationship with a company and user, because plans/prices belong to apps, companies can subscribe to apps, so you can have plans/prices that companies are not using so not relationship